src/XPurchase/VirtoCommerce.XPurchase/Authorization/CanAccessCartAuthorizationRequirement .cs

```diff
@@ -20,25 +21,31 @@ public class CanAccessCartAuthorizationHandler : PermissionAuthorizationHandlerB
     {
         protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, CanAccessCartAuthorizationRequirement requirement)
         {
-            var result = false;
-            if (context.Resource is ShoppingCart cart)
-            {
-                result = cart.CustomerId == GetUserId(context);
-            }
-            else if (context.Resource is SearchCartQuery searchQuery)
-            {
-                searchQuery.UserId = GetUserId(context);
-                result = true;
-            }
-            if (result)
-            {
-                context.Succeed(requirement);
-            }
-            else
+
+            var result = context.User.IsInRole("__administrator");
+
+            if (!result)
             {
-                context.Fail();
-            }
+                if (context.Resource is ShoppingCart cart)
+                {
+                    result = cart.CustomerId == GetUserId(context);
+                }
+                else if (context.Resource is SearchCartQuery searchQuery)
+                {
+                    searchQuery.UserId = GetUserId(context);
+                    result = true;
+                }
 
+                if (result)
+                {
+                    context.Succeed(requirement);
+                }
+                else
+                {
+                    context.Fail();
+                }
```

Comment:
Doesn't it need to be done outside the if clause - for the admin we need to succeed a requirement?

Reply:
context.HasSucceeded is true by default. But, yes. Better to move if clause outside